### PR TITLE
Implement initial version of team list on team tab

### PR DESCRIPTION
**Pull request by akalin-keybase**

### CODE REVIEW

File: shared/teams/index.js [hunk 1]
```diff
@@ -3,7 +3,25 @@ import Render from './render'
 import pausableConnect from '../util/pausable-connect'
 import openURL from '../util/open-url'
 
-const mapStateToProps = state => ({})
+import type {TypedState} from '../constants/reducer'
+
+const mapStateToProps = (state: TypedState) => {
+  // TODO: Figure out better way to get list of teams -- this won't
+  // work until the inbox is loaded.
```
Comment (akalin-keybase) [inline shared/teams/index.js]: It sounds like I'll have to plumb up the team member count from the service, so can probably put all the info we need there too

**Comment by mmaxim** [comment]
For list of teams, you can just call the `TeamList` RPC on the service.

File: shared/teams/render.js [hunk 2]
```diff
@@ -69,13 +72,45 @@ const BetaNote = (props: BetaNoteProps) => (
   </Box>
 )
 
-type Props = {
-  onCreateTeam: () => void,
-  onJoinTeam: () => void,
-  onReadDoc: () => void,
+type TeamListProps = {
+  // TODO: Change to map to member count.
+  teams: I.Set<Teamname>,
```
Comment (chrisnojima) [inline shared/teams/render.js]: we should avoid passing immutable to the dumb components, lets convert it in mergeProps

File: shared/teams/render.js [hunk 3]
```diff
@@ -85,11 +120,12 @@ const Render = (props: Props) => (
         ...globalStyles.flexBoxColumn,
         alignItems: 'center',
         marginBottom: globalMargins.tiny,
-        marginLeft: globalMargins.large,
-        marginRight: globalMargins.large,
+        marginLeft: globalMargins.tiny,
```
Comment (chrisnojima) [inline shared/teams/render.js]: see my banner pr, the margins likely shouldn't be here, scrolling gets messed up in mobile

File: shared/teams/index.js (hunk 1 above)
Comment (akalin-keybase) [reply shared/teams/index.js]: Done, hooked up to teamList RPC

File: shared/teams/index.js [hunk 4]
```diff
@@ -3,7 +3,25 @@ import Render from './render'
...
+  const inbox = state.chat.get('inbox')
+  const teams = {}
+  inbox.forEach(i => {
```
[the remark this answers is not in the document]
Comment (akalin-keybase) [reply shared/teams/index.js]: moot

File: shared/teams/index.js [hunk 5]
```diff
@@ -3,7 +3,25 @@ import Render from './render'
...
+    if (i.teamname) {
+      teams[i.teamname] = {}
+    }
+  })
+  let teamNames = Object.keys(teams)
```
[the remark this answers is not in the document]
Comment (akalin-keybase) [reply shared/teams/index.js]: done

File: shared/teams/render.js [hunk 6]
```diff
@@ -62,7 +62,37 @@ export const BetaNote = (props: BetaNoteProps) => (
...
+              flex: 1,
+              marginRight: globalMargins.tiny,
+            }}
+          >
+            <Avatar size={32} teamname={info.name} />
+            <Text type="BodySemibold" style={{flex: 1, marginLeft: globalMargins.tiny}}>
+              {info.name}
+            </Text>
+          </Box>
+          {isMobile ? null : <Divider style={{marginLeft: 44}} />}
```
[the remark this answers is not in the document]
Comment (akalin-keybase) [reply shared/teams/render.js]: done

File: shared/teams/render.js (hunk 2 above)
Comment (akalin-keybase) [reply shared/teams/render.js]: done

File: shared/teams/render.js (hunk 3 above)
Comment (akalin-keybase) [reply shared/teams/render.js]: done, changed to padding in TeamList

**Comment by akalin-keybase** [comment]
Comments addressed, PTAL

**Pull request merged**